src/vs/workbench/browser/parts/editor/editorGroupsControl.ts

```diff
@@ -2013,6 +2129,32 @@ export class EditorGroupsControl extends Themable implements IEditorGroupsContro
 			}
 		});
 
+		// Layout centered Editor
+		const doCentering =
+			this.layoutVertically &&
+			this.visibleSilos === 1 &&
+			this.partService.isCenteredLayoutActive() &&
+			this.visibleEditors[Position.ONE] instanceof TextResourceEditor;
```

Comment:
This `instanceof` check is definetely not needed, it makes the behavior of this feature unpredictable, and I am not sure what is the reason for this limiation